doc/managing_pachyderm/backup_restore_and_migrate.md

```diff
@@ -0,0 +1,410 @@
+The above cloning/snapshotting technique is recommended when doing a migration where minimizing downtime is desirable, as it allows the duplicated object store to be the basis of the upgraded, new cluster instead of requiring Pachyderm to extract the data from object store.
+
+### 3. Restart all pipeline and data loading operations
+
+Once the backup is complete, restart all paused pipelines and data loading operations.
+
+From the directed acyclic graphs (DAG) that define your pachyderm cluster, start each pipeline.    You can either run a multiline shell command, shown below, or you must, for each pipeline, manually run the `pachctl start-pipeline` command.
+
+`pachctl start-pipeline pipeline-name`
+
+You can confirm each pipeline is started using the pachctl list-pipeline command
+
+`pachctl list-pipeline`
+
+A useful shell script for running `start-pipeline` on all pipelines is included below.  It may be necessary to install the utilities used in the script, like `jq` and `xargs`, on your system.
+
+```
+pachctl list-pipeline --raw \
+  | jq -r '.pipeline.name' \
+  | xargs -P3 -n1 -I{} pachctl start-pipeline {}
+```
+
+If you used the port-changing technique, [above](#pausing-data-loading-operations), to stop all data loading into Pachyderm from outside processes, you should change the ports back.
+
+```
+# Once you have restarted all running pachyderm pipelines, such as with this command,
+# $ pachctl list-pipeline --raw \
+#   | jq -r '.pipeline.name' \
+#   | xargs -P3 -n1 -I{} pachctl start-pipeline {}
+
+# all pipelines in your cluster should be restarted. To restart all data loading 
+# processes, we're going to change the pachd Kubernetes service so that
+# it only accepts traffic on port 30650 again (from 30649). 
+#
+# Backup the Pachyderm service spec, in case you need to restore it quickly
+$ kubectl get svc/pach -o json >pach_service_backup_30649.json
+
+# Modify the service to accept traffic on port 30650, again
+$ kubectl get svc/pachd -o json | sed 's/30649/30650/g' | kc apply -f -
+
+# Modify your environment so that *you* can talk to pachd on the old port
+$ export PACHD_ADDRESS="${PACHD_ADDRESS/30649/30650}"
+
+# Make sure you can talk to pachd (if not, firewall rules are a common culprit)
+$ pc version
+COMPONENT           VERSION
+pachctl             1.7.11
+pachd               1.7.11
+```
+## General restore procedure
+### Restore your backup to a pachyderm cluster, same version
+
+Spin up a Pachyderm cluster and run `pachctl restore` with the backup you created earlier.
+
+`pachctl restore < path/to/your/backup/file`
+
+You can also use the `-u` or `--url` flag to get the backup directly from the object store you placed it in
+
+`pachctl restore --url s3://...`
+
+## General migration procedure
+
+### Types of Migrations
+You'll need to follow one of two procedures when upgrading a Pachyderm cluster:
+
+1. Simple migration path (for minor versions, e.g. v1.8.2 to v1.8.3)
```

Comment:
Numbering messed up?

Reply:
That's just the way that markdown does numbered lists.  You always start them with 1. and md does the rest

1. Just
1. Like 
1. This